lesson03.find-a-pair/script.js

```diff
@@ -0,0 +1,95 @@
+const card = document.getElementsByClassName("mem-card");
+const game = document.getElementsByClassName("container")[0];
+const scoreDis = document.getElementById("score");
+let allCards = [...card];
+let array = ["A", "A", "B", "B", "C", "C", "D", "D", "E", "E", "F", "F", "G", "G", "H", "H"];
+let flipCounter = 0;
+let score = 0;
+let flipedCards = [];
+let bothCardFliped = false;
+let tries = 0;
+
+for (let i = 0; i < allCards.length; i++) {
+    allCards[i].addEventListener("click", displayCard);
+    allCards[i].id = array[i]
+    allCards[i].addEventListener("click", flipCards);
+}
+// shuffling the cards function 
+function shuffle(a) {
+    let j, x, i;
+    for (i = a.length - 1; i > 0; i--) {
+        j = Math.floor(Math.random() * (i + 1));
+        x = a[i];
+        a[i] = a[j];
+        a[j] = x;
+    }
+    return a;
+}
+// Shuffle at web load
+shuffle(allCards);
+// function for display the card 
+function displayCard() {
+    if (bothCardFliped) return;
+    this.classList.toggle("flip");
+}
+// flip Cards function
+function flipCards() {
+    if (flipCounter === 0) {
+        // firstCard = this;
+        flipedCards.push(this);
+        flipCounter++;
+    }
+    else if (flipCounter === 1) {
+        flipedCards.push(this);
+        flipCounter === 0;
+        checkMatch();
+    }
+}
+// check if match cards
+function checkMatch() {
+    if (flipedCards[0].id === flipedCards[1].id) {
+        flipedCards[0].removeEventListener("click", displayCard);
```

Comment:
No need to remove the event, use a flag (boolean variable) to prevent action while checking match